Formula/vivid.rb

```diff
@@ -0,0 +1,29 @@
+class Vivid < Formula
+  desc "Generator for LS_COLORS with support for multiple color themes"
+  homepage "https://github.com/sharkdp/vivid"
+  url "https://github.com/sharkdp/vivid/archive/v0.5.0.tar.gz"
+  sha256 "d4864169dca54e44c1f3a6f09bcf5048190e6a97ee0a9887437dbc7f5ed6aaaa"
+
+  depends_on "rust" => :build
+
+  def install
+    ENV.append_to_cflags "-fno-stack-check" if DevelopmentTools.clang_build_version >= 1010
+    system "cargo", "install", "--locked", "--root", prefix, "--path", "."
+    # Update when https://github.com/sharkdp/vivid/blob/master/ci/before_deploy.bash changes.
+    pkgshare.install "config/filetypes.yml", "themes"
+  end
+
+  # Hardcoded to /usr/share/vivid or $HOME/.config/vivid, doesn't check XDG_CONFIG_HOME.
+  def caveats
+    <<~EOS
+      To use Vivid you need to link the vivid share directory to ~/.config/vivid by running:
+
+        mkdir -p ~/.config
+        ln -sf #{HOMEBREW_PREFIX}/share/vivid ~/.config/vivid
+    EOS
+  end
+
+  test do
+    assert_match "vivid #{version}", shell_output("#{bin}/vivid --help")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Sounds good, I'm blocked on this by https://github.com/Homebrew/homebrew-core/pull/55962#discussion_r437355908 but will tackle it afterwards.